src/engine/QueryExecutionTree.cpp

```diff
@@ -193,8 +123,10 @@ QueryExecutionTree::selectedVariablesToColumnIndices(
     }
   } else {
     for (auto var : selectedVarsOrAsterisk.getSelectVariables()) {
-      if (var.starts_with("TEXT(")) {
-        var = var.substr(5, var.rfind(')') - 5);
+      constexpr std::string_view prefix = "TEXT(";
+      constexpr size_t prefixLength = prefix.length();
+      if (var.starts_with(prefix)) {
+        var = var.substr(prefixLength, var.rfind(')') - prefixLength);
```

Comment:
// TODO: The `TEXT(?variable)` syntax is redundant and will probably removed when we have  a proper SPARQL parser.